core/src/main/java/io/questdb/std/str/Path.java

```diff
@@ -134,32 +126,26 @@ public Path concat(CharSequence str) {
     }
 
     public Path concat(long pUtf8NameZ) {
-
         ensureSeparator();
-
         long p = pUtf8NameZ;
         while (true) {
-
-            if (len + OVERHEAD >= capacity) {
-                extend(len * 2 + OVERHEAD);
-            }
-
             byte b = Unsafe.getUnsafe().getByte(p++);
-            if (b == 0) {
+            if (b == NULL) {
                 break;
             }
 
-            Unsafe.getUnsafe().putByte(wptr, (byte) (b == '/' && Os.type == Os.WINDOWS ? '\\' : b));
-            wptr++;
-            len++;
+            int requiredCapacity = length();
+            if (requiredCapacity + OVERHEAD >= capacity) {
+                extend(requiredCapacity * 2 + OVERHEAD);
+            }
+            Unsafe.getUnsafe().putByte(tailPtr++, (byte) (b == '/' && Os.isWindows() ? '\\' : b));
         }
-
         return this;
     }
 
     public Path concat(CharSequence str, int from, int to) {
         ensureSeparator();
-        copy(str, from, to);
+        encodeUtf8(str, from, to);
```

Comment:
We have to have  UTF8 encoding explicit. It is not free and we don't have to pay the cost everywhere.

Lets create new method

Reply:
copy had this signature: 
```
private void copy(CharSequence str, int from, int to) {
        encodeUtf8(str, from, to);
    }
```

what do you mean?